src/charm.py

```diff
@@ -71,6 +169,40 @@ def _on_config_changed(self, _):
                 value = self.model.config[key]
                 logger.info("Setting {} to: {}".format(key, value))
                 self._stored.config[key] = value
+
+        # this should go into install hook
+        uid = pwd.getpwnam("root").pw_uid
+        gid = grp.getgrnam("root").gr_gid
+
+        path = "/etc/nginx/ssl"
+        if not os.path.exists(path):
+            os.makedirs(path, 0o755)
+            os.chown(path, uid, gid)
+            os.chmod(path, 0o755)
+
+        config = self._stored.config
+        self._stored.config["ssl_enabled"] = (
+            "ssl_cert" in config.keys()  # noqa: W503
+            and "ssl_key" in config.keys()  # noqa: W503
+            and "ssl_cert" in self._stored.config.keys()  # noqa: W503
+            and "ssl_key" in self._stored.config.keys()  # noqa: W503
+        )
+
+        # write ssl certificate if present
+        if "ssl_cert" in config.keys() and self._stored.config["ssl_cert"]:
+            cert_path = os.path.join("/etc/nginx/ssl", "server.crt")
+            cert = b64decode(self._stored.config["ssl_cert"])
+            write_file(cert_path, cert, "root", "root", 0o644)
+
+        if "ssl_key" in config.keys() and self._stored.config["ssl_key"]:
+            key_path = os.path.join("/etc/nginx/ssl", "server.key")
+            key = b64decode(self._stored.config["ssl_key"])
+            write_file(key_path, key, "root", "root", 0o640)
+
+        if "ca_cert" in config.keys() and self._stored.config["ca_cert"]:
+            ca_cert = self._stored.config["ca_cert"]
+            install_ca_cert(b64decode(ca_cert), "nginx-server.crt")
```

Comment:
suggestion: this seems repetitive; we have already checked that all three configs are set in the paragraph above, so this could be rewritten as (pseudocode)
```
if (all three configs are set):
    self._stored.config["ssl_enabled"] = True
    write the three files
```